compaction/compaction_manager.cc

```diff
@@ -1615,15 +1615,18 @@ future<> compaction_manager::perform_cleanup(owned_ranges_ptr sorted_owned_range
         throw std::runtime_error("cleanup request failed: sorted_owned_ranges is empty");
     }
 
-    co_await run_with_compaction_disabled(t, [this, &t, sorted_owned_ranges] () -> future<> {
-        auto update_sstables_cleanup_state = [&] (const sstables::sstable_set& set) {
-            return set.for_each_sstable_gently([&] (const sstables::shared_sstable& sst) {
-                update_sstable_cleanup_state(t, sst, *sorted_owned_ranges);
+    bool found_maintenance_sstables;
+    co_await run_with_compaction_disabled(t, [this, &t, sorted_owned_ranges, &found_maintenance_sstables] () -> future<> {
```

Comment:
I think `run_with_compaction_disabled()` should allow return value pass-through and this lambda would become much simpler.

Reply:
It's orthogonal to this PR, but sounds good.